python/cudf/cudf/core/indexing.py

```diff
@@ -103,21 +103,33 @@ def __setitem__(self, key, value):
         self._sr.iloc[key] = value
 
     def _loc_to_iloc(self, arg):
+        from cudf.core.column import column
         from cudf.core.series import Series
         from cudf.core.index import Index
 
+        if is_scalar(arg):
+            try:
+                found_index = self._sr.index._values.find_first_value(
+                    arg, closest=False
+                )
+                return found_index
+            except (TypeError, KeyError, IndexError, ValueError):
+                raise IndexError("label scalar is out of bound")
+
         if isinstance(arg, (cudf.MultiIndex, pd.MultiIndex)):
             if isinstance(arg, pd.MultiIndex):
                 arg = cudf.MultiIndex.from_pandas(arg)
 
             return indices_from_labels(self._sr, arg)
 
-        if isinstance(
-            arg, (list, pd.Series, range, Index, DeviceNDArray, np.ndarray)
-        ) or hasattr(arg, "__cuda_array_interface__"):
+        if (
+            isinstance(arg, (list, pd.Series, range, Index, DeviceNDArray))
```

Comment:
Should we just consider making this the `else` clause where if it's not a scalar, a slice, or a MultiIndex we call `as_column` on it and construct a Series from that? It centralizes the logic of what can be translated into a column as opposed to redefining it in a bunch of places.